velox/expression/CastExpr.cpp

```diff
@@ -335,14 +335,33 @@ VectorPtr CastExpr::applyMap(
       nestedRows, elementToTopLevelRows, oldErrors);
   context.swapErrors(oldErrors);
 
+  // Returned map vector should be addressable for every element, even those
+  // that are not selected.
+  BufferPtr sizes = input->sizes();
+  if (newMapKeys->isConstantEncoding() && newMapValues->isConstantEncoding()) {
+    // We extends size since that is cheap.
+    newMapKeys->resize(input->mapKeys()->size());
+    newMapValues->resize(input->mapValues()->size());
+
+  } else if (
+      newMapKeys->size() < input->mapKeys()->size() ||
+      newMapValues->size() < input->mapValues()->size()) {
+    sizes =
+        AlignedBuffer::allocate<vector_size_t>(rows.end(), context.pool(), 0);
+    auto* inputSizes = input->rawSizes();
+    auto* rawSizes = sizes->asMutable<vector_size_t>();
+    rows.applyToSelected(
+        [&](vector_size_t row) { rawSizes[row] = inputSizes[row]; });
+  }
+
   // Assemble the output map
   return std::make_shared<MapVector>(
       context.pool(),
       MAP(toType.keyType(), toType.valueType()),
       input->nulls(),
       rows.end(),
       input->offsets(),
```

Comment:
Do we need to do the same for offsets as well?

Reply:
not needed since when the size is 0 offsets are dont care.

Reply:
Discussed with Laith offline. I'm good with this code.